src/docs/cursor-ops.dox

```diff
@@ -161,4 +161,42 @@ that may not be modified or freed by the application.  If a longer scope
 is required, the application must make a copy of the memory before the
 cursor is re-used, closed or reset.
 
- */
+@section cursor_bound Apply bounds to the cursor
+
+A cursor may have a set of bounds applied to it, this restricts the key range
+that the cursor operates within.
+
+Suppose we have a table with keys \c A to \c Z. We can restrict the cursor to only operate over
+keys \c L->Q. If the cursor walks past key \c Q, the cursor will enter an unpositioned state having
+walked off the "end" of the table from its perspective. Additionally should the cursor perform a
+search_near, it will only search within the key range of \c L->Q.  This can provide an application with a
+performance boost when operating within tables that have a large number of deleted or invisible records.
+This is because the key comparison with the bound occurs before we determine if the value associated
+with the key is visible or not. The cursor can have a lower bound set or an upper bound set, or both.
+
+@snippet ex_cursor.c cursor bound
+
+Both or either one of the lower and upper bounds can be set, and any unset bound will be assumed to be off the start or
```

Comment:
I would rephrase this a bit: Both or either one of the lower and upper bounds can be set, if a bound is not set then the usual start or end of the file applies. Given our previous example had we set a lower bound of \c L and no upper bound then our key range would be \c L->Z.